Dynamic parameters to marathonfile

Hi, thanks for the library!

Could you please clarify if it's possible to pass dynamic parameters to marathon file config somehow?
For instance, for filteringConfiguration. (For one case I'm going to run all tests, for another - only specific ones)

Or the only way for now it's always generate new marathonfile by custom codegen?

@Malinskiy  Tried this but failed. 
Command line:
`STRATEGY="device-model" /Users/abhishek.bedi/Downloads/marathon-0.7.3/bin/marathon`

MarathonFile:
`name: "sample-ios-app tests"
outputDir: "derived-data/Marathon"
ignoreFailures: true
testClassRegexes: ["^((?!Abstract).)*Tests$"]
poolingStrategy:
  type: "#{STRATEGY}"
batchingStrategy:
  type: "fixed-size"
  size: 1
debug: true
filteringConfiguration:
  allowlist:
    - type: "fully-qualified-test-name"
      values:
        - "sample-appUITests.StoryboardTests#testButton"
        - "sample-appUITests.StoryboardTests#testLabel"
vendorConfiguration:
  type: "iOS"
  derivedDataDir: "derived-data"
  sourceRoot: "sample-appUITests"
  knownHostsPath: ${HOME}/.ssh/known_hosts
  remoteUsername: ${USER}
  remotePrivateKey: ${HOME}/.ssh/marathon
`
**Output:**

` ~/Downloads/marathon-develop/sample/ios-app  STRATEGY="device-model" /Users/abhishek.bedi/Downloads/marathon-0.7.3/bin/marathon
10:02:18.263 [main] INFO  c.m.marathon.cli.ApplicationView - Starting marathon v0.7.3
10:02:18.383 [main] INFO  c.m.m.e.BugsnagExceptionsReporter - Init BugSnag
10:02:18.383 [main] INFO  c.m.marathon.cli.ApplicationView - Checking Marathonfile config
10:02:19.420 [main] INFO  c.m.m.e.BugsnagExceptionsReporter - Finish BugSnag
10:02:19.420 [main] DEBUG com.bugsnag.Bugsnag - Closing connection to Bugsnag
Exception in thread "main" com.malinskiy.marathon.config.exceptions.ConfigurationException: Error parsing config file /Users/abhishek.bedi/Downloads/marathon-develop/sample/ios-app/Marathonfile
	at com.malinskiy.marathon.config.serialization.ConfigurationFactory.parse(ConfigurationFactory.kt:81)
	at com.malinskiy.marathon.cli.ApplicationViewKt$main$1.invoke(ApplicationView.kt:42)
	at com.malinskiy.marathon.cli.ApplicationViewKt$main$1.invoke(ApplicationView.kt:29)
	at com.xenomachina.argparser.SystemExitExceptionKt.mainBody(SystemExitException.kt:74)
	at com.xenomachina.argparser.SystemExitExceptionKt.mainBody$default(SystemExitException.kt:72)
	at com.malinskiy.marathon.cli.ApplicationViewKt.main(ApplicationView.kt:27)
Caused by: com.fasterxml.jackson.databind.exc.InvalidTypeIdException: Could not resolve type id '#{STRATEGY}' as a subtype of `com.malinskiy.marathon.config.strategy.PoolingStrategyConfiguration`: known type ids = [abi, combo, device-model, manufacturer, omni, os-version] (for POJO property 'poolingStrategy')
 at [Source: (StringReader); line: 7, column: 9]
	at com.fasterxml.jackson.databind.exc.InvalidTypeIdException.from(InvalidTypeIdException.java:43)
	at com.fasterxml.jackson.databind.DeserializationContext.invalidTypeIdException(DeserializationContext.java:2073)
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownTypeId(DeserializationContext.java:1564)
	at com.fasterxml.jackson.databind.jsontype.impl.TypeDeserializerBase._handleUnknownTypeId(TypeDeserializerBase.java:298)
	at com.fasterxml.jackson.databind.jsontype.impl.TypeDeserializerBase._findDeserializer(TypeDeserializerBase.java:165)
	at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer._deserializeTypedForId(AsPropertyTypeDeserializer.java:125)
	at com.fasterxml.jackson.databind.jsontype.impl.AsPropertyTypeDeserializer.deserializeTypedFromObject(AsPropertyTypeDeserializer.java:110)
	at com.fasterxml.jackson.databind.deser.AbstractDeserializer.deserializeWithType(AbstractDeserializer.java:263)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:539)
	at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer._deserializeUsingPropertyBased(BuilderBasedDeserializer.java:411)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1405)
	at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.deserializeFromObject(BuilderBasedDeserializer.java:319)
	at com.fasterxml.jackson.databind.deser.BuilderBasedDeserializer.deserialize(BuilderBasedDeserializer.java:219)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:322)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4674)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3629)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3597)
	at com.malinskiy.marathon.config.serialization.ConfigurationFactory.parse(ConfigurationFactory.kt:42)
	... 5 more
10:02:19.423 [Thread-2] DEBUG com.bugsnag.Bugsnag - Closing connection to Bugsnag
`

--- a/docs/_posts/2018-11-19-configuration.md
+++ b/docs/_posts/2018-11-19-configuration.md
@@ -1710,6 +1710,47 @@ marathon {
 
 See relevant vendor module page, e.g. [Android][3] or [iOS][4]
 
+# Dynamic configuration
+
+
+## Marathonfile
+Marathon supports envvar interpolation in the Marathonfile: every occurance of `${X}` in the Marathonfile will be replaced with the value of
+envvar `X`, for example:
+
+{% tabs dynamic-configuration-yaml %} {% tab dynamic-configuration-yaml Marathonfile %}
+
+```yaml
+outputConfiguration:
+  maxPath: ${MARATHON_MAX_PATH}
+```
+
+{% endtab %} {% endtabs %}
+
+## Gradle
+
+You can also provide dynamic configuration for gradle configuration using programming language syntax, for example:
+{% tabs dynamic-configuration-gradle %} {% tab dynamic-configuration-gradle Gradle %}
+
+```kotlin
+marathon {
+  outputConfiguration {
+    maxPath = System.properties['maxpath'] as int
+  }
+}
+```
+
+{% endtab %} {% tab dynamic-configuration-gradle Gradle Kotlin %}
+
+```kotlin
+marathon {
+  outputConfiguration {
+    maxPath = someWayOfRetrievingTheParameter()
+  }
+}
+```
+
+{% endtab %} {% endtabs %}
+
 [1]: https://www.influxdata.com/
 
 [2]: https://graphiteapp.org/

